templates.js

```diff
@@ -1,1857 +1,1848 @@
+          default: 'FormView'
+        },
+        {
+          id: 'fieldLabelMap',
+          type: 'text',
+          displayName: 'Map of HTML form field names to human friendly labels',
+          description: 'Optional: Form analytics will detect field labels attached to every field and it will ' +
+            'send them to the server for easier identification of fields in reports. However, if your page supports' +
+            ' different language versions the data may become fragmented. To fix this issue, configure JSON map of ' +
+            'HTML field names to labels that should be used in reports or set your preferred names using ' +
+            '"Analytics" > "Settings" > "Dimension value grouping". ' +
+            'Example: `{"name": "First name", "surname": "Last name"}`.',
+          default: '{}'
+        },
+        {
+          id: 'iframeTracking',
+          type: 'boolean',
+          displayName: 'Send messages from iframes',
+          description: 'If checked, you will pass messages to the parent window instead of _paq.push',
+          default: false
+        },
+      ],
+    },
+    {
+      id: 'trackCopiedText',
+      name: 'Track copied text',
+      description: `
+      This template allows you to track pieces of text that are copied to clipboard by your website users.
+      `,
+      template: `
+  ${fs.readFileSync(path.join(__dirname, 'build/postIframeMessage.js'), { encoding: 'utf-8' })}    
+  ${fs.readFileSync(path.join(__dirname, 'build/pushToAnalytics.js'), { encoding: 'utf-8' })}        
+  ${fs.readFileSync(path.join(__dirname, 'build/trackCopiedText.js'), { encoding: 'utf-8' })}
+  
+  trackCopiedText(function (eventData) {
+    var trackFromIframe = {{iframeTracking}}; 
+    if(!trackFromIframe){
+      pushToAnalytics(eventData);
+    } else {
+      postIframeMessage(eventData);
+    }
+  }, {
+    filteredElements: ['input', 'textarea']
+  });
+      `,
+      arguments: [
+        { id: 'iframeTracking',
+        type: 'boolean',
+        displayName: 'Send messages from iframes',
+        description: 'If checked, you will pass messages to the parent window instead of _paq.push',
+        default: false
+        },      
+     ],
+    },
+    {
+      id: 'iframeTrackingHandler',
+      name: 'Iframe Tracking Handler',
+      description: `
+        This is an iframe tracking handler.
+        Add one of the templates with iframe tracking turned on to the website
+        that will be shown as an iframe and use this handler to catch the messages
+        sent from the iframe on the parent website and turn them into events.
+      `,
+      template: `
+  ${fs.readFileSync(path.join(__dirname, 'build/pushToAnalytics.js'), { encoding: 'utf-8' })}   
+  
+  window.addEventListener('message', function(event){ 
+    if (event.origin !== 'https://'+'{{domain}}') {
+      return;
+    }

```

Comment:
What is decision behind it?

Reply:
https://piwikpro.atlassian.net/browse/PPPSYS-48527

"Any script in a document in a window can request to let a message get dispatched to a document in another window whose window object it has obtained, by calling .postMessage() on that window object. Consequently, any event listener used to receive messages must first check the identity of the sender of the message, using the origin and possibly source properties. This cannot be overstated: Failure to check the origin and possibly source properties enables cross-site scripting attacks."